src/Enum.php

```diff
@@ -8,213 +8,246 @@
     }
 
-    public function __call($name, $arguments)
+    public static function make($value): Enumerable
     {
-        if (strlen($name) > 2 && strpos($name, 'is') === 0) {
-            return $this->equals(substr($name, 2));
-        }
+        if (is_int($value)) {
+            $index = $value;
 
-        if (isset(self::resolve()[strtolower($name)])) {
-            return static::__callStatic($name, $arguments);
+            if (! static::isIndex($index)) {
+                throw new InvalidIndexException($value, static::class);
+            }
+
+            return new static(array_search($index, static::toArray()), $index);
+        } elseif (is_string($value)) {
+            $value = strtolower($value);
+
+            if (! static::isValue($value)) {
+                throw new InvalidValueException($value, static::class);
+            }
+
+            if (method_exists(static::class, $value)) {
+                return forward_static_call(static::class.'::'.$value);
+            }
+
+            return new static($value, static::toArray()[$value]);
         }
 
-        throw new \BadMethodCallException(sprintf('Call to undefined method %s->%s()', static::class, $name));
+        throw new TypeError(sprintf('%s::make() expects string|int as argument but %s given', static::class, gettype($value)));
     }
 
-    /**
-     * @param string|\Spatie\Enum\Enum $enum
-     *
-     * @return bool
-     */
-    public function equals($enum): bool
+    public static function isIndex(int $index): bool
     {
-        if (is_string($enum)) {
-            $enum = static::from($enum);
-        }
-
-        if (! $enum instanceof $this) {
-            return false;
-        }
+        return in_array($index, static::getIndices());
+    }
 
-        if ($enum->value !== $this->value) {
-            return false;
-        }
+    public static function isValue(string $value): bool
+    {
+        return in_array($value, static::getValues());
+    }
 
-        return true;
+    public static function getIndices(): array
+    {
+        return array_values(static::toArray());
     }
 
-    /**
-     * @param string[]|\Spatie\Enum\Enum[] $enums
-     *
-     * @return bool
-     */
-    public function isOneOf(array $enums): bool
+    public static function getValues(): array
     {
-        foreach ($enums as $enum) {
-            if ($this->equals($enum)) {
-                return true;
-            }
-        }
+        return array_keys(static::toArray());
+    }
 
-        return false;
+    public static function toArray(): array
+    {
+        return static::resolve();
     }
 
-    public function __toString(): string
+    public function getValue(): string
     {
         return $this->value;
     }
 
-    public function jsonSerialize()
+    public function getIndex(): int
     {
-        return $this->value;
+        return $this->index;
     }
 
-    public static function toArray(): array
+    public function isEqual($value): bool
```

Comment:
Why changing `equals` to `isEqual`?

Reply:
I've changed this to match all the other `isXYZ` methods. Was a consistency decision.

Reply:
Fair enough!